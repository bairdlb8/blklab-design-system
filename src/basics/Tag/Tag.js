import React from 'react'
import PropTypes from 'prop-types'

import './Tag.scss'

const Tag = ({ className = '', children }) => {
  return <div className="ds-tag">{children}</div>
}

Tag.propTypes = {}

export default Tag
