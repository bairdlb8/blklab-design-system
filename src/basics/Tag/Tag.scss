@import '../../scss/_mixins';

div.ds-tag {
  display: inline-flex;
  padding: var(--spacing-1, 2px) var(--spacing-4, 1em);
  background-color: var(--gray-color-light);
  color: var(--gray-color-darker);
  border-radius: 5px;

  @include selectFontColor(var(--text-color-default));

  &.dark {
  }

  &.accent {
  }
}
