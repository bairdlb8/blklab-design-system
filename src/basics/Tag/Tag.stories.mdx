import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks'

import Tag from './Tag'

<Meta title="Basics / Tag" component={ Tag } />

# Tag

## Add some text here

export const Template = (args) => {
  return <Tag {...args} />
}

<Preview withToolbar>
  <Story name="Primary" args={ {} }>
    {Template.bind({})}
  </Story>
</Preview>

<Props of={ Tag } />
