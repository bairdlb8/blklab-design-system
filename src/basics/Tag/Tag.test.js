import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { toHaveNoViolations } from 'jest-axe'
import Tag from './Tag'

expect.extend(toHaveNoViolations)

describe('<Tag />', () => {

  describe('accessibility', () => {
    it('should not fail any accessibility tests', async () => {
      const { container } = render(<Tag />)
      expect(await axe(container)).toHaveNoViolations()
    })
  })

  it('should take a snapshot', () => {
    const { asFragment } = render(<Tag />)

    expect(asFragment(<Tag />)).toMatchSnapshot()
  })

})
