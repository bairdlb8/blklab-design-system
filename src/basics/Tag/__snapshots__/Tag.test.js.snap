// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    Tag
  </div>
</DocumentFragment>
`;
